core/src/main/java/org/keycloak/representations/idm/authorization/AuthorizationSchema.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.representations.idm.authorization;
+
+import java.util.Set;
+
+public interface AuthorizationSchema {
```

Comment:
I have a feeling that in the future the schema will be a `class` and created by implementations of some SPI provider. But for now, it does not matter much ...

Perhaps simpler now is to just have `AuthorizationSchema` as a class and have the `FineGrainedAdminPermissionsAuthorizationSchema` like that:

```
class FineGrainedAdminPermissionsAuthorizationSchema extends AuthorizationSchema {

     public static final FineGrainedAdminPermissionsAuthorizationSchema INSTANCE = new FineGrainedAdminPermissionsAuthorizationSchema(); 

     private FineGrainedAdminPermissionSchema() {
         super(new ResourceType("Users", new HashSet<>(Arrays.asList("manage"))));
     }
}
```

Wdyt?

Reply:
Thank you! I see where you're coming from, but I personally don't find this approach significantly simpler at this point. According to previous discussions, it seemed to me we should aim for an interface for `AuthorizationSchema`. 

However, if you foresee that `AuthorizationSchema` will be a `class` I’ll to adjust accordingly now. Let me know if that aligns with what you have in mind!

Reply:
Updated according to your suggestion.

Reply:
It does not matter much now :) It is an implementation detail, at least for now. 

It should make things simpler and consistent because we won't have sub-types of the schema but instances of the schema with different configurations.